### PR TITLE
Fix deserialization issue with unknown properties, fix media type issue (restlet/apispark#3771)

### PR DESCRIPTION
**Pull request by quilicicf**

### CODE REVIEW

File: modules/org.restlet.ext.apispark/src/org/restlet/ext/apispark/internal/conversion/swagger/v1_2/SwaggerUtils.java [hunk 1]
```diff
@@ -90,19 +138,22 @@ public static Definition getDefinition(String swaggerUrl, String userName,
...
                         "Reading file: " + swaggerUrl + api.getPath());
                 apis.put(
                         api.getPath(),
-                        createAuthenticatedClientResource(
-                                swaggerUrl + api.getPath(), userName, password)
-                                .get(ApiDeclaration.class));
+                        getAndDeserialize(swaggerUrl + api.getPath(), userName, password, ApiDeclaration.class));
+            }
+            try {
+                client.stop();
```
Comment (thboileau) [inline modules/org.restlet.ext.apispark/src/org/restlet/ext/apispark/internal/conversion/swagger/v1_2/SwaggerUtils.java]: should be the last instruction executed in the whole method.

File: modules/org.restlet.ext.apispark/src/org/restlet/ext/apispark/internal/conversion/swagger/v1_2/SwaggerTranslator.java [hunk 2]
```diff
@@ -586,20 +587,26 @@ private static void fillContract(Contract contract,
...
+            contract.setDescription(listing.getInfo().getDescription());
+        }
+
         LOGGER.log(Level.FINE, "Contract " + contract.getName() + " added.");
-        contract.setDescription(listing.getInfo().getDescription());
         definition.setContract(contract);
 
-        if (definition.getEndpoints().isEmpty()) {
+        if (definition.getEndpoints().isEmpty()
+                && listing.getAuthorizations() != null) {
```
Comment (boillodmanuel) [inline modules/org.restlet.ext.apispark/src/org/restlet/ext/apispark/internal/conversion/swagger/v1_2/SwaggerTranslator.java]: why not check listing.getAuthorizations() != null after having create the endpoint (does it make sens to have an endpoint without authentication info? for me yes)

Comment (boillodmanuel) [inline modules/org.restlet.ext.apispark/src/org/restlet/ext/apispark/internal/conversion/swagger/v1_2/SwaggerTranslator.java]: I don't know why we check that definition.getEndpoints().isEmpty()??

File: modules/org.restlet.ext.apispark/src/org/restlet/ext/apispark/internal/conversion/swagger/v1_2/SwaggerUtils.java [hunk 3]
```diff
@@ -55,9 +62,17 @@
     protected static Logger LOGGER = Logger.getLogger(SwaggerUtils.class
             .getName());
 
+    private static Client client = new Client(Arrays.asList(Protocol.HTTP, Protocol.HTTPS));
+
+    static {
+        client.setContext(new Context());
+        client.getContext().getParameters().add("idleTimeout", "4000");
```
Comment (boillodmanuel) [inline modules/org.restlet.ext.apispark/src/org/restlet/ext/apispark/internal/conversion/swagger/v1_2/SwaggerUtils.java]: default value is 60000. Is it too high?

File: modules/org.restlet.test/src/org/restlet/test/ext/apispark/conversion/swagger/v2_0/SwaggerLoader.java [hunk 4]
```diff
@@ -27,17 +27,29 @@
...
 
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.wordnik.swagger.models.Swagger;
 import com.wordnik.swagger.util.Json;
 
 public class SwaggerLoader {
 
+    private static ObjectMapper mapper() {
+        ObjectMapper mapper = Json.mapper();
```
Comment (boillodmanuel) [inline modules/org.restlet.test/src/org/restlet/test/ext/apispark/conversion/swagger/v2_0/SwaggerLoader.java]: have you check the source code or the javadoc? I think that mapper() return a static instance

**Pull request merged**